### DISCUSSION TITLE
ax.set_aspect('equal')

### DISCUSSION THREAD

**Discussion by jiajinbu** [mwaskom/seaborn#3076] [Q&A]
Does seaborn object can make the scale of x and y equal, like `ax.set_aspect('equal')`.

**Comment by liuzj039** [comment]
Same issue. I have tried use following scripts, but still not work.

```python
fig = (
    so.Plot(penguins, x="bill_length_mm", y="bill_depth_mm")
    .add(so.Dot())
    .plot()._figure
)
fig.axes[0].set_aspect("equal")
fig
```
![image](https://user-images.githubusercontent.com/55271666/195037364-c0d38c8b-0a25-4b6a-a729-69a931f4cc61.png)

**Comment by mwaskom** [comment]
@liuzj039 why do you think that's not working? Looks right to me. Here's a pure matplotlib version:

```python
f, ax = plt.subplots()
ax.scatter(x="bill_length_mm", y="bill_depth_mm", data=penguins)
ax.set_aspect("equal")
```
![image](https://user-images.githubusercontent.com/315810/195070640-58df23a8-2063-478a-aba6-235e4f6f01a3.png)

In any case, this functionality isn't exposed through the `Plot` interface yet, to answer the main question.

**Reply by liuzj039** [reply]
> @liuzj039 why do you think that's not working? Looks right to me. Here's a pure matplotlib version:
> 
> ```python
> f, ax = plt.subplots()
> ax.scatter(x="bill_length_mm", y="bill_depth_mm", data=penguins)
> ax.set_aspect("equal")
> ```
> 
> ![image](https://user-images.githubusercontent.com/315810/195070640-58df23a8-2063-478a-aba6-235e4f6f01a3.png)
> 
> In any case, this functionality isn't exposed through the `Plot` interface yet, to answer the main question.

I seem to have misunderstood the meaning of aspect. The code below is fine.

```
fig = (
    so.Plot(penguins, x="bill_length_mm", y="bill_depth_mm")
    .add(so.Dot())
    .layout()
    .limit(x=(0,60), y=(0, 60))
    .plot()._figure
)
fig.axes[0].set_aspect("equal")
fig
```
![image](https://user-images.githubusercontent.com/55271666/195077889-ace82cc2-4aba-4d90-acd6-866475b2d0de.png)

**Comment by mwaskom** [comment]
Yes, `aspect="equal"` means that the same distance in data coordinates along the two axes will take up the same distance in physical coordinates. So you only get a square plot if the limits of the axes are identical.

**Comment by liuzj039** [comment]
Thank you very much for your detailed explanation
